sycl/plugins/opencl/pi_opencl.cpp

```diff
@@ -342,6 +384,103 @@ pi_result piDeviceGetInfo(pi_device device, pi_device_info paramName,
     return PI_SUCCESS;
   }
 
+  case PI_DEVICE_INFO_PARTITION_PROPERTIES: {
+    // SYCL spec says: if this SYCL device cannot be partitioned into at least
+    // two sub devices then the returned vector must be empty.
+    pi_uint32 partitionMaxSubDevices = 0;
+    if (device->subLevel == -1)
+      device->subLevel = getSubLevel(device);
+    if (device->isRootDevice()) {
+      clGetDeviceInfo(
+          cast<cl_device_id>(device), CL_DEVICE_PARTITION_MAX_SUB_DEVICES,
+          sizeof(partitionMaxSubDevices), &partitionMaxSubDevices, nullptr);
+    } else if (device->isSubDevice()) {
+      // find out number of CCSes
+      bool supported = false;
+      cl_int ret_err = CL_SUCCESS;
+      ret_err =
+          checkDeviceExtensions(cast<cl_device_id>(device),
+                                {"cl_intel_command_queue_families"}, supported);
+      if (ret_err != CL_SUCCESS)
+        return static_cast<pi_result>(ret_err);
+      if (!supported) {
+        std::cout
+            << "This device does not support cl_intel_command_queue_families"
+            << std::endl;
+        return ReturnValue(pi_device_partition_property{0});
```

Comment:
I realize that the same pattern is used everywhere else, but I think that we should be using a named constexpr variable for this value in `pi.h` instead. Would be ok if not addressed in this PR though (as we'd need to change other plugins as well).

Reply:
On the other hand, I'm not sure why we are returning `info::partition_property::no_partition`. Shouldn't it be an empty vector instead. I guess I have the same question about Level Zero implementation, so might be I'm just missing something.

Reply:
I think it might be a '0' terminated list.

Reply:
Will prefer to make that change in a separate PR.